File: corchetes_balanceados/corchetes_balanceados.py
class CorchetesBalanceados:

    @staticmethod
    def run(string: str) -> bool:
        corchetes_open = 0
        for char in string:
            if char == '[':
                corchetes_open += 1
            elif char == ']':
                corchetes_open -= 1
            if corchetes_open < 0:
                return False
        if corchetes_open != 0:
            return False
        return True


if __name__ == "__main__":
    cadena = input("Ingrese el string a comprobar: ")
    print(CorchetesBalanceados.run(cadena))

File: corchetes_balanceados/corchetes_balanceados.go
package main
import "fmt"

func corchetesBalanceados(cadena string) bool {
    corchetes_open := 0;
    for i := 0 ; i < len(cadena); i++ {
       
        char := cadena[i]
        if char == '[' {
            corchetes_open += 1
        } else if char == ']' {
            corchetes_open -= 1
        }
        if corchetes_open < 0 {
            return false
        }
    } 
    if corchetes_open != 0 {
        return false
    }
    return true
}

func main() {
    fmt.Println(corchetesBalanceados("]"))
}
